openshift/deploy/deploymentconfig/postgres-master.yml

```diff
@@ -60,16 +76,27 @@ objects:
                   value: 2GB
                 - name: POSTGRESQL_EFFECTIVE_CACHE_SIZE
                   value: 8GB
+                - name: PGDATA
+                  value: /var/lib/pgsql/data/userdata
               image: ${OC_REGISTRY}/${OC_PROJECT}/${PREFIX}postgres:${GIT_SHA1}
               imagePullPolicy: IfNotPresent
+              lifecycle:
+                preStop:
+                  exec:
+                    command:
+                      - /usr/bin/env
+                      - bash
+                      - -c
+                      - |
+                        pg_ctl stop;
```

Comment:
if we don't nicely stop postgres, we quickly end up with a corrupt datastore and postgres refuses to restart. this is also documented...somewhere...